course/views.py

```diff
@@ -33,17 +33,33 @@ def get_common_objects(self):
         self.welcome_text = settings_text('WELCOME_TEXT')
         self.internal_user_label = settings_text('INTERNAL_USER_LABEL')
         self.external_user_label = settings_text('EXTERNAL_USER_LABEL')
-        self.instances = []
-        prio2 = []
-        treshold = timezone.now() - datetime.timedelta(days=10)
-        for instance in CourseInstance.objects.get_visible(self.request.user)\
-                .filter(ending_time__gte=timezone.now()):
-            if instance.starting_time > treshold:
-                self.instances += [instance]
-            else:
-                prio2 += [instance]
-        self.instances += prio2
-        self.note("welcome_text", "internal_user_label", "external_user_label", "instances")
+        self.my_instances = []
+        self.all_instances = []
+        end_threshold = timezone.now() - datetime.timedelta(days=30)
+        user = self.request.user
+
+        if user and user.is_authenticated:
+            for course in user.userprofile.teaching_courses.all():
```

Comment:
Yeah, the teacher-part was trickier because it went through the Course so I couldn't find any other way to do it as simply as with the other two categories through the reverse relation. So in the end ended up doing it the way I found it here with the CachedTopMenu: https://github.com/apluslms/a-plus/blob/master/course/cache/menu.py. I am not sure is it a problem there as well? Maybe the caching is a discussion for another day :)  

And in any case, now that I see there is a readymade related field on the CourseInstance-level as well, that seems like a really good fix. Will do

Reply:
> I am not sure is it a problem there as well? Maybe the caching is a discussion for another day :)

Yes, it should be done better. However, it isn't so worrisome as the cache entry is updated only very rarely. On the other hand, the front page is updated in every reload. Technically, we could use the cache to populate the front page too, but it's likely that the drop menu and the front page will have a bit different set of information or the details would require database query anycase.

Reply:
Yes, from what I gather they require a bit different set of information. But in any case, good to know that about the cache and how important this issue there.